redata/statistics/tests.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+from scipy.stats import chi2
+
+class Mahalanobis_distance():
+
+    """
+        Description:
+        ------------
+
+        Class that computes the mahalanobis distance in order to detect multidimensional outliers.
+        For more info, check: 
+
+            https://en.wikipedia.org/wiki/Mahalanobis_distance
+            https://blogs.sas.com/content/iml/2012/02/15/what-is-mahalanobis-distance.html
+        
+        Parameters:
+        -----------
+
+            means_vector [optional numpy means vector]
+            cov [optional numpy covariance matrix]
+            p_value [optional float]: probability of obtaining test results at least as extreme as the results actually
+                             observed, under the assumption that the null hypothesis is correct
+
+                              H0: mu_i = mu_j / i != j
+                              H1: mu_i != mu_j
+
+                              * mu_i -> mean of i
+                             
+                             All the records that have a p-value lower than self.p_value will be detected as outlier
+        
+        Important:
+        ----------
+
+        This distance only works with 2 or more variables. In case of using 1 variable it will not work and it should be
+        fine using the z_score.
+    """
+
+    def __init__(self, means_vector=None, cov_matrix=None, p_value=0.05):
+        
+        self.means_vector = means_vector
```

Comment:
I get means_vector , cov_matrix for computing it possibly more efficiently but I wouldn't assume you pass this into class (You most likely can _cache if inside class etc) But for now maybe just remove those 2 and assume it will be computed later